### ISSUE TITLE
Baseline model: k-mer frequency + XGBoost

### ISSUE THREAD

**Issue by smodee** [smodee/pathogen-classifier#8]
## Summary

Build a baseline classifier using k-mer frequency features + XGBoost. This establishes a performance floor before investing in transformer training, and validates the full train/evaluate pipeline.

**Priority: high | No dependencies | Can run in parallel with #9, #10**

## Implementation

Create `src/baseline_model.py` with:

### Key functions
- `compute_kmer_frequencies(sequence, k=5) → dict` — count all k-mer occurrences in a sequence
- `extract_features(sequences, k=5) → np.ndarray of shape (n, 4^k)` — normalized k-mer frequency vectors (k=5 → 1024 features)
- `train_baseline(train_path, val_path, k, output_dir, ...)` — full pipeline: load CSVs, extract features, train XGBoost with class-weighted sample weights, evaluate on val set
- `predict_baseline(test_path, model_dir) → (y_true, y_pred, y_prob)` — load saved model and predict

### CLI
```
python src/baseline_model.py --train data/processed/train.csv --val data/processed/val.csv
```

### Outputs
Saves to `models/baseline/`:
- `xgboost_model.json` — trained model
- `label_encoder.json` — family → index mapping
- `kmer_params.json` — k value and feature ordering
- `val_metrics.json` — validation set metrics

### Class imbalance handling
Use XGBoost `sample_weight` parameter with inverse-frequency class weights. Poxviridae dominates at 56% of training chunks (see issue #6).

### Dependencies
Add `xgboost>=2.0.0` to `requirements.txt`.

## Tests

Create `tests/test_baseline_model.py`:
- k-mer frequency vector shape and normalization (sums to ~1.0)
- Class weights computation
- Mini train/predict integration test with synthetic data
- Saved model file existence after training

## Existing code to reuse
- `FAMILY_ORDER` from `src/data_exploration.py` for consistent label ordering
- CSV format: `id, parent_sequence_id, chunk_index, sequence, length, family` (from `data/processed/train.csv`)

## Acceptance criteria
- [ ] `python src/baseline_model.py --train data/processed/train.csv --val data/processed/val.csv` completes without error
- [ ] Model files saved to `models/baseline/`
- [ ] `val_metrics.json` contains accuracy, macro F1, weighted F1, balanced accuracy, per-class metrics
- [ ] All tests pass: `pytest tests/test_baseline_model.py -v`
